match_all_ids.py

```diff
@@ -0,0 +1,45 @@
+import os
+import sys
+
+currpath = str(os.getcwd())
+sys.path.append(currpath+"/../")
+
+from normalize_license_text.normalize_class import NormalizeText
+from compare_template_text.normalize_template_text import NormalizeTemplate
+from normalize_license_text.compare_normalized_files import CompareNormalizedFiles
+
+input_license_text = str(sys.argv[1])
+
+directory = 'data_templates/'
+
+"""This file on executing runs through all the License Templates and returns 
+the matched IDs. """
+
+try:
+    with open(input_license_text,'r') as inputfile:
+        input_text_string = inputfile.read()
+        inputfile.close()
+        x = NormalizeText(input_text_string)
+        normalized_text_string = x.returnfinalstring_for_template()
+
+except IOError:
+    print("There is no file named ",input_license_text)
+
+for filename in os.scandir(directory):
+    print(filename.path)
+    try:
+        with open(filename.path,'r') as input_file:
+            input_template_file = input_file.read()
+            input_file.close()
+            object_normalization = NormalizeText(input_template_file)
+            input_template_file = object_normalization.returnfinalstring_for_template()
+            
+            y = NormalizeTemplate(normalized_text_string, input_template_file)
+            y.normalize_template()
+            normalized_template_string = y.return_normalized_template()
+    except:
+        continue
+        
```

Comment:
Did you add tests for this?

Reply:
yes

Reply:
@anshuldutt21 I am confused, can you tell the file name/ test name for this?

Reply:
I have added tests for the template text comparison, which can be found in the tests folder with the name test_compare_template_text.py. The file match_all_ids.py just compares all the template files with the given text for which the tests are already included in the test_compare_template_text.py file.

Reply:
I think you need to write a separate test for this. That test is specific to `template text comparison`, the tests for these would be different, since it returns a totally different output. Also, please add a name for which the template matched.

Reply:
@rtgdk I am not sure as to how to write tests for the match_all_ids file. The file is just a python script which loops for all the templates in the data and matches each template against the input license text according to the Normalize_Template class which I have implemented and tested earlier in the compare_license_text folder.

Reply:
@anshuldutt21 The test would be testing whether your `match_all_ids` functions works as expected or not, or throws error as expected. It would be completely different from the basic test you have written for a particular template matching case.